server/src/server.ts

```diff
@@ -81,7 +87,10 @@ connection.onDidChangeConfiguration((change) => {
   const settings = change.settings as Settings
   bitBakeProjectScanner.shouldDeepExamine = settings.bitbake.shouldDeepExamine
   logger.level = settings.bitbake.loggingLevel
-  bitBakeProjectScanner.pathToBuildFolder = settings.bitbake.pathToBuildFolder
+  // eslint-disable-next-line no-template-curly-in-string
+  const pathToBuildFolder = settings.bitbake.pathToBuildFolder.replace('${workspaceFolder}', workspaceRoot)

```

Comment:
Seems like ${workspaceFolder} is only resolved by vscode itself in certain contexts but not in a case like here. Feels unnatural to have this code but I don't see a better way either...

Reply:
I expect the python and bash files to become much bigger. Also we'll appreciate not to work on the same files if we eventually have to share the tasks for embedded python and embedded bash (less conflicts).

I also plan to move the types from utils into the "common" package. Having the remaining functions in the same file with general-support would create a circular dependency.